002-es6-modules.md

```diff
@@ -89,9 +89,53 @@ When `require()`ing a file.
 
 ### Determining if source is an ES6 Module
 
-~~A new filetype will be recognised, `.jsm` as ES6 based modules. They will be treated as a different loading semantic but compatible with existing systems, just like `.node`, `.json`, or usage of `require.extension` (even though deprecated) are compatible. It would be ideal if we could register the filetype with IANA as an offical file type, see [TC39 issue](https://github.com/tc39/ecma262/issues/322). Though it seems this would need to go through the [IESG](https://www.ietf.org/iesg/) and it seems browsers are non-plussed on introducing a new MIME.~~ Both TC39 and WHATWG reject the notion of a new MIME, we would have to push this through without any support from the standards bodies. As such I am for the moment disregarding but keeping reference to this idea.
+A new filetype will be recognised, `.jsm` as ES6 based modules. They will be treated as a different loading semantic but compatible with existing systems, just like `.node`, `.json`, or usage of `require.extension` (even though deprecated) are compatible. It would be ideal if we could register the filetype with IANA as an offical file type, see [TC39 issue](https://github.com/tc39/ecma262/issues/322). Though it seems this would need to go through the [IESG](https://www.ietf.org/iesg/) and it seems browsers are non-plussed on introducing a new MIME.
 
-We will be using a [Directive Prologue](https://tc39.github.io/ecma262/#directive-prologue) in order to detect if source code is ES6. The directive will be `"use module"`.
+The `.jsm` file extension will have a higher loading priority than `.js`.
+
+### ES6 Import Resolution
+
+ES6 `import` statements will not perform non-exact searches on relative or absolute paths, unlike `require()`. This means that no file extensions, or index files will be searched for when using relative or absolute paths. `node_modules` based paths will continue to use searching for both compatibility and to not limit the ability to have `package.json` support both ES6 and CJS entry points in a single codebase.
```

Comment:
Does this really mean that `import` statements will now always require an explicit file extension to be included, or am I mis-reading it?